dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dxf2/events/EventImportTest.java

```diff
@@ -244,67 +246,137 @@ protected void setUpTest()
+        programStageA.setProgram( programWithRegistration );
+        programStageA2.setProgram( programWithRegistration );
+        programWithRegistration.getProgramStages().add( programStageA );
+        programWithRegistration.getProgramStages().add( programStageA2 );
         programStageB.getProgramStageDataElements().add( programStageDataElementB );
-        programStageB.setProgram( programB );
-        programB.getProgramStages().add( programStageB );
+        programStageB.setProgram( programWithoutRegistration );
+        programWithoutRegistration.getProgramStages().add( programStageB );
         manager.update( programStageA );
         manager.update( programStageA2 );
-        manager.update( programA );
+        manager.update( programWithRegistration );
         manager.update( programStageB );
-        manager.update( programB );
-        pi = new ProgramInstance();
-        pi.setEnrollmentDate( new Date() );
-        pi.setIncidentDate( new Date() );
-        pi.setProgram( programB );
-        pi.setStatus( ProgramStatus.ACTIVE );
-        pi.setStoredBy( "test" );
-        pi.setName( "EventImportTestPI" );
-        pi.setUid( CodeGenerator.generateUid() );
-        manager.save( pi );
+        manager.update( programWithoutRegistration );
+        programInstance = new ProgramInstance();
+        programInstance.setEnrollmentDate( new Date() );
+        programInstance.setIncidentDate( new Date() );
+        programInstance.setProgram( programWithoutRegistration );
+        programInstance.setStatus( ProgramStatus.ACTIVE );
+        programInstance.setStoredBy( "test" );
+        programInstance.setName( "EventImportTestPI" );
+        programInstance.setUid( CodeGenerator.generateUid() );
+        manager.save( programInstance );
         event = createEvent( "eventUid001" );
-        createUserAndInjectSecurityContext( true );
     }
 
     @Test
     void testAddEventOnProgramWithoutRegistration()
         throws IOException
     {
-        InputStream is = createEventJsonInputStream( programB.getUid(), programStageB.getUid(),
+        InputStream is = createEventJsonInputStream( programWithoutRegistration.getUid(), programStageB.getUid(),
             organisationUnitB.getUid(), null, dataElementB, "10" );
         ImportSummaries importSummaries = eventService.addEventsJson( is, null );
         assertEquals( ImportStatus.SUCCESS, importSummaries.getStatus() );
     }
 
     @Test
-    void testAddEventWithDueDateForProgramWithoutRegistration()
+    void shouldSetEnrollmentLastUpdatedUserInfoWhenAddEventOnProgramWithRegistration()
+        throws IOException
     {
-        String eventUid = CodeGenerator.generateUid();
+        Enrollment enrollment = createEnrollment( programWithRegistration.getUid(),
+            trackedEntityInstanceMaleA.getTrackedEntityInstance() );
 
-        Enrollment enrollment = createEnrollment( programA.getUid(),
+        assertEquals( ImportStatus.SUCCESS,
```

Comment:
I understand the purpose of this assertion is to stop the execution of the test in case the enrollment is not imported correctly, but the goal of this test is to verify whether the field lastUpdatedByUser is updated or not, not that the enrollment or the event below are imported correctly. Should we take for granted that the import will work and there's already a test somewhere doing this validation? Or maybe create a new class and have this imports done during set up?